python/pyspark/sql/connect/plan.py

```diff
@@ -2214,7 +2214,9 @@ def __init__(
         self._input_grouping_cols = input_grouping_cols
         self._other_grouping_cols = other_grouping_cols
         self._other = cast(LogicalPlan, other)
-        self._func = function._build_common_inline_user_defined_function(*cols)
+        # The function takes entire DataFrame as inputs, no need to do
+        # column binding (no input columns).
+        self._func = function._build_common_inline_user_defined_function()
```

Comment:
cc @zhengruifeng @HyukjinKwon

Reply:
it seems this field is not used in connect

https://github.com/apache/spark/blob/48e207f4a2192d474f2c0f141b984ef0c36a78c3/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala#L694-L723

cc @xinrong-meng

Reply:
yea, that's why this bug is hidden. We simply picked the first matching dataframe in the query plan to resolve columns, which is wrong and now we will throw AMBIGUOUS_COLUMN_REFERENCE